drivers/ma_yya_followup_report/app/models/ma_yya_followup_report/report.rb

```diff
@@ -0,0 +1,86 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaYyaFollowupReport
+  class Report
+    include ArelHelper
+    include Filter::FilterScopes
+
+    def initialize(filter)
+      @filter = filter
+      @end_date = @filter.on
+      @start_date = @end_date - 2.months - 3.weeks
+      @late_date = @end_date - 3.months
+    end
+
+    attr_accessor :start_date, :late_date
+
+    def clients
+      client_scope.pluck(*columns.values).
+        map do |row|
+          Hash[columns.keys.zip(row)]
+        end.
+        sort_by { |row| row[:last_seen] }
+    end
+
+    def columns
+      window = ::Arel::Nodes::Window.new.partition(c_t[:id])
+      {
+        id: :id,
+        first_name: :FirstName,
+        last_name: :LastName,
+        engagement_date: she_t[:first_date_in_program].minimum.over(window),
+        last_seen: cls_t[:InformationDate].maximum.over(window),
+      }
+    end
+
+    private def client_scope
+      client_with_enrollment_scope.
+        left_outer_joins(service_history_enrollments: [enrollment: :current_living_situations]).
+        where.not(id: client_with_contact_scope.select(:id))
+    end
+
+    private def client_with_enrollment_scope
+      ::GrdaWarehouse::Hud::Client.
+        distinct.
+        joins(:service_history_enrollments).
+        merge(enrollment_scope)
+    end
+
+    private def client_with_contact_scope
+      ::GrdaWarehouse::Hud::Client.
+        distinct.
+        joins(service_history_enrollments: [enrollment: :current_living_situations]).
+        merge(enrollment_scope).
+        merge(contact_scope)
+    end
+
+    private def enrollment_scope
+      scope = ::GrdaWarehouse::ServiceHistoryEnrollment.
+        entry.
+        open_between(start_date: @start_date, end_date: @end_date)
```

Comment:
Any reason not to use:
```suggestion
      scope = ::GrdaWarehouse::ServiceHistoryEnrollment.entry
      scope = filter_for_range(scope)
```